archlinuxcn/video-trimmer/PKGBUILD

```diff
@@ -0,0 +1,36 @@
+# Maintainer: torculus <hidden> 
+# Contributor: Mark Wagie <mark dot wagie at proton dot me>
+pkgname=video-trimmer
+pkgver=0.8.2
+pkgrel=1
+pkgdesc="Trim videos quickly"
+arch=('x86_64')
+url="https://apps.gnome.org/VideoTrimmer"
+license=('GPL3')
+depends=('ffmpeg' 'gst-plugins-good' 'libadwaita')
+makedepends=('blueprint-compiler' 'cargo' 'git' 'meson')
+checkdepends=('appstream-glib')
+source=("https://gitlab.gnome.org/YaLTeR/video-trimmer/-/archive/v$pkgver/$pkgname-v$pkgver.tar.gz")
+sha256sums=('6bada3db4a553a2dab543e18228233262f3ab177e73de2360e17236a13c51c5f')
+
+prepare() {
+  cd "$pkgname-v$pkgver"
+  export CARGO_HOME="$srcdir/cargo-home"
+  export RUSTUP_TOOLCHAIN=stable
+  cargo fetch --target "$CARCH-unknown-linux-gnu"
+}
+
+build() {
+  export CARGO_HOME="$srcdir/cargo-home"
+  export RUSTUP_TOOLCHAIN=stable
+  arch-meson "$pkgname-v$pkgver" build
+  meson compile -C build
+}
+
+check() {
+  meson test -C build --print-errorlogs
+}
+
+package() {
+  meson install -C build --destdir "$pkgdir"
+}
```

Comment:
```
video-trimmer W: ELF file ('usr/bin/video-trimmer') lacks GNU_PROPERTY_X86_FEATURE_1_SHSTK.
video-trimmer W: Unused shared library '/usr/lib/libpango-1.0.so.0' by file ('usr/bin/video-trimmer')
video-trimmer W: Unused shared library '/usr/lib/libgdk_pixbuf-2.0.so.0' by file ('usr/bin/video-trimmer')
video-trimmer W: Unused shared library '/usr/lib/libcairo.so.2' by file ('usr/bin/video-trimmer')
```

Reply:
This is quit difficult and I couldn't find a good answer.

1. ELF lacks GNU_PROPERTY_X86_FEATURE_1_SHSTK:
[Detect missing GNU_PROPERTY_X86_FEATURE_1_SHSTK in ELF files.](https://gitlab.archlinux.org/pacman/namcap/-/issues/63)
[namcap reports GNU_PROPERTY_X86_FEATURE_1_SHSTK, what action to take?](https://bbs.archlinux.org/viewtopic.php?id=292194)

2. Unused shared library:
[Unused shared libraries](https://github.com/TokTok/c-toxcore/issues/359)

After reading those, still don't how to fix. I think need some helps.

Reply:
Please report this issue to the upstream.

Reply:
> Please report this issue to the upstream.

reported [here](https://gitlab.gnome.org/YaLTeR/video-trimmer/-/issues/72)